bolt/lib/Passes/PAuthGadgetScanner.cpp

```diff
@@ -169,16 +186,26 @@ struct State {
   std::vector<SmallPtrSet<const MCInst *, 4>> LastInstWritingReg;
   State() {}
   State(unsigned NumRegs, unsigned NumRegsToTrack)
-      : NonAutClobRegs(NumRegs), LastInstWritingReg(NumRegsToTrack) {}
-  State &operator|=(const State &StateIn) {
-    NonAutClobRegs |= StateIn.NonAutClobRegs;
+      : SafeToDerefRegs(NumRegs), LastInstWritingReg(NumRegsToTrack) {}
+
+  /// Returns S, so that S.merge(S1) == S1.merge(S) == S1.
+  static State getMergeNeutralElement(unsigned NumRegs,
+                                      unsigned NumRegsToTrack) {
+    State S(NumRegs, NumRegsToTrack);
+    S.SafeToDerefRegs.set();
```

Comment:
I'm finding it somewhat counter-intuitive that the initial state is that all registers are considered safe to dereference. Maybe this needs a comment to explain why that is the case? (I think that intuitively, at first sight, it seems to make more sense if all registers were considered unsafe to dereference?)

(Maybe it's more obvious if you think about this method as merging the neutral element, and the neutral element is a state in which all registers are safe to dereference?
Anyway, I think this method could use a comment to explain this as I find it not entirely trivial)